src/network/networkd-dhcp4.h

```diff
@@ -17,13 +20,29 @@ typedef enum DHCPClientIdentifier {
         _DHCP_CLIENT_ID_INVALID = -1,
 } DHCPClientIdentifier;
 
+struct DHCPOption {
+        Network *network;
+        NetworkConfigSection *section;
+
+        sd_dhcp_option *option;
+
+        LIST_FIELDS(DHCPOption, options);
```

Comment:
Hmm, is it really necessary to define this structure? Why not put use an array of `sd_dhcp_option` directly in Network?

Reply:
Not to duplicate the memory . Lib is different.

Reply:
I don't understand this. Please explain.

Reply:
It's same implementation ad ```Prefix``` and stored in a ```set``` inside the lib. size of ```sd_dhcp_option``` is not constant also . In a array how can we store variable length ? In the memory saving context these are stored in .network only the pointer is passed o the library. When the link goes down we don't free the pointer just the set and no duplicate of memory.

Reply:
In `Network`, add
`sd_dhcp_option **options; size_t n_options; size_t options_allocated`
and when adding a new option, construct the `sd_dhcp_option` structure, and move it to the array with:
```C
struct sd_dhcp_option *option = ...;
GREEDY_REALLOC(network->options, network->options_allocated, network->n_options + 1);
network->options[network->n_options++] = option;
```